src/sys/unix.rs

```diff
@@ -67,6 +67,15 @@ pub const IPPROTO_TCP: i32 = libc::IPPROTO_TCP;
 pub const IPPROTO_UDP: i32 = libc::IPPROTO_UDP;
 pub const SOCK_SEQPACKET: i32 = libc::SOCK_SEQPACKET;
 pub const SOCK_RAW: i32 = libc::SOCK_RAW;
+pub const IP_ADD_SOURCE_MEMBERSHIP: libc::c_int = 39;
+
+#[repr(C, packed)]
+#[derive(Copy, Clone)]
+pub struct IpMreqSource {
```

Comment:
These should be pushed up into the libc crate.

Reply:
yeah.  those i figured would have to upstream.  any other thoughts?